src/components/input/Select.tsx

```diff
@@ -607,37 +454,18 @@ function SelectMain<T>({
           listboxOverflow,
         }}
       >
-        <div
-          className={classnames(
-            'absolute z-5 max-h-80 overflow-y-auto overflow-x-hidden',
-            'rounded border bg-white shadow hover:shadow-md focus-within:shadow-md',
-            listboxAsPopover && [
-              // We don't want the listbox to ever render outside the viewport,
-              // and we give it a 16px gap
-              'max-w-[calc(100%-16px)]',
-              // Overwrite [popover] default styles
-              'p-0 m-0',
-            ],
-            !listboxAsPopover && {
-              // Hiding instead of unmounting to
-              // * Ensure screen readers detect button as a listbox handler
-              // * Listbox size can be computed to correctly drop up or down
-              hidden: !listboxOpen,
-              'right-0': alignListbox === 'right',
-              'min-w-full': true,
-            },
-            listboxClasses,
-          )}
-          ref={popoverRef}
-          // nb. Use `undefined` rather than `false` because Preact doesn't
-          // handle boolean values correctly for this attribute (it will set
-          // `popover="false"` instead of removing the attribute).
-          popover={listboxAsPopover ? 'auto' : undefined}
-          data-testid="select-popover"
+        <Popover
+          anchorElementRef={wrapperRef}
```

Comment:
This change is not related with the extraction of `Popover` specifically (this would have been `anchorElementRef={buttonRef}`, but I realized the border is set in the container, not the button, which makes popover positioning and sizing calculation fail if it's done relative to the button.

It is easier to see the issue with a thicker border.

Using the button as the reference:
![Captura desde 2024-11-21 10-21-42](https://github.com/user-attachments/assets/ce72ffa9-c70b-42be-b3ff-bbb9ed3fb25b)

Using the wrapper as the reference:
![Captura desde 2024-11-21 10-21-13](https://github.com/user-attachments/assets/1d33443e-2e74-497b-8f88-91031ecd87df)